backend/asm_targets/asm_directives_intf.ml

```diff
@@ -0,0 +1,240 @@
+      DWARF debugging information standard). *)
+  val uleb128 : ?comment:string -> Numbers.Uint64.t -> unit
+
+  (** Emit a 64-bit integer in signed LEB128 variable-length encoding. *)
+  val sleb128 : ?comment:string -> Int64.t -> unit
+
+  (** Emit a string (directly into the current section). This function does not
+      write a terminating null. *)
+  val string : ?comment:string -> string -> unit
+
+  (** Cache a string for later emission. The returned label may be used to
+      obtain the address of the string in the section. This function does not
+      emit anything. (See [emit_cached_strings], below.) If a string is supplied
+      to this function that is already in the cache (associated with the same
+      section and comment) then the previously-assigned label is returned, not a
+      new one. *)
+  val cache_string : ?comment:string -> Asm_section.t -> string -> Asm_label.t
+
+  (** Emit the sequence of: label definition: <string><null terminator> pairs as
+      per previous calls to [cache_string] with appropriate directives to switch
+      section interspersed. This function clears the cache. *)
+  val emit_cached_strings : unit -> unit
+
+  (** Emit a comment. *)
+  val comment : string -> unit
+
+  (** Emit a blank line. *)
+  val new_line : unit -> unit
+
+  (** Define a data ("object") symbol at the current output position. When
+      emitting for MASM this will cause loads and stores to/from the symbol to
+      be treated as if they are loading machine-width words (unless the
+      instruction has an explicit width suffix). *)
+  val define_data_symbol : Asm_symbol.t -> unit
+
+  (** Mark a symbol as global. *)
+  val global : Asm_symbol.t -> unit
+
+  (** Emit a machine-width reference to the given symbol. *)
+  val symbol : ?comment:string -> Asm_symbol.t -> unit
+
+  (** Define a label at the current position in the current section. The
+      treatment for MASM when emitting into non-text sections is as for
+      [define_data_symbol], above. *)
+  val define_label : Asm_label.t -> unit
+
+  (** Emit a machine-width reference to the given label. *)
+  val label : ?comment:string -> Asm_label.t -> unit
+
+  (** Emit a machine-width reference to the address formed by adding the given
+      byte offset to the address of the given symbol. The symbol may be in a
+      compilation unit and/or section different from the current one. *)
+  val symbol_plus_offset : Asm_symbol.t -> offset_in_bytes:Targetint.t -> unit
+
+  (** Emit a machine-width reference giving the displacement between two given
+      symbols. To obtain a positive result the symbol at the [lower] address
+      should be the second argument, as for normal subtraction. The symbols must
+      be in the current compilation unit and in the same section. *)
+  val between_symbols_in_current_unit :
+    upper:Asm_symbol.t -> lower:Asm_symbol.t -> unit
+
+  (** Like [between_symbols], but for two labels, emitting a 16-bit-wide
```

Comment:
(Should probably be `[between_symbols_in_current_unit]`,
multiple occurrences below.)